From 782fa5ba3965c0a4d7bc399fb72cb55a974dc55d Mon Sep 17 00:00:00 2001
From: Rohan Puri <rohan_puri@xyratex.com>
Date: Thu, 24 Oct 2013 12:09:52 +0530
Subject: [PATCH 100/107] Changes summary : - 1. FIXED Bug: - On memory
 failure, fop should be made NULL, after freeing its internal data ptrs.   
 This should be done in caller of addb_mon_fop_free() & not in
 addb_mon_fop_free(), since we do    not pass address of fops pointer to this
 function.

This BUG was found while reading of posting summaries code today.
---
 addb/addb_monitor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/addb/addb_monitor.c b/addb/addb_monitor.c
index c04beb2..b9eb383 100644
--- a/addb/addb_monitor.c
+++ b/addb/addb_monitor.c
@@ -437,7 +437,6 @@ static void addb_mon_fop_free(struct m0_stats_update_fop *fop_data,
 		m0_free(stats->sf_stats);
 	}
 	m0_free(fop_data);
-	fop_data = NULL;
 }
 
 static void addb_monitor_stats_fop_release(struct m0_ref *ref)
@@ -526,6 +525,7 @@ M0_INTERNAL int m0_addb_monitor_summaries_post(struct m0_reqh       *reqh,
 		    data == NULL) {
 			result = -ENOMEM;
 			addb_mon_fop_free(fop, stats_nr);
+			fop = NULL;
 			break;
 		}
 		rec = &fop->suf_stats.sf_stats[used++];
-- 
1.8.3.2

